onnx/onnx.in.proto

```diff
@@ -422,6 +422,10 @@ message TypeProto {
     // This field MUST be present for this version of the IR.
     optional TensorProto.DataType elem_type = 1;
     optional TensorShapeProto shape = 2;
+    // An optional denotation can be used to denote the whole 
+    // tensor with a standard semantic description as to what is 
+    // stored inside
+    optional string denotation = 3;
```

Comment:
We'd put this denotation in TypeProto instead of Tensor so that other types (say Map, Sequence) will also be able to use it?

Reply:
Yes, I had thought of that option as well.   I think the real challenge here is that we have Tensor as our base …. and all of these other things are sub classes of tensor.   Map and Sequence are first class citizens right now.   I imagine after this experiment, we would promote Image/Audio/Text to first class citezens as well.    Thus my though was to keep it on tensor to allow us to experiment for candidates to promote to first class types.    
Also, maps and sequences don't have the denotations we define in this first iteration.   A map cannot be an image/audio/text.   Nor can a sequence.   I think if we find candidates for denotation on map & sequence (Things that are common to all TypeProto's) ; we could add one then.
 thoughts?